[MIG][9.0] template: Upgrade JS API

lasley opened the pull request:
### Milestone (Odoo version)
- 
  v9
### Module(s)
- 
  template
### Fixes / new features
- Upgrade JS to new API
- Upgrade onload method to use core bus web_client_ready
- Add an exported method

REF https://github.com/OCA/web/pull/386#discussion_r72024916

cc @antespi @pedrobaeza

pedrobaeza commented:
Thanks for this update

:+1:

antespi commented:
:+1: Thanks @lasley

pedrobaeza commented:
@moylop260, there's a strange error in Travis. Do you have an idea of it?

moylop260 commented:
I saw the same error from stable branch build
https://travis-ci.org/OCA/maintainer-tools/builds/147765216

It is a traceback from flake8

I'll use travis2docker to debug and fix it..

moylop260 commented:
FYI I can reproduce it with t2d using the travis original image (with other environment I can't reproduce it)

The version with problem flake8 3.0.2
<img width="879" alt="screen shot 2016-07-29 at 9 27 07 pm" src="https://cloud.githubusercontent.com/assets/6644187/17267624/8582df84-55d3-11e6-9911-92cb66932bea.png">

I install a old version 2.6.2 and work fine!
<img width="856" alt="screen shot 2016-07-29 at 9 29 05 pm" src="https://cloud.githubusercontent.com/assets/6644187/17267623/856b2e20-55d3-11e6-9c88-4afce487c54e.png">

I will create a new PR with a freeze of version

obulkin commented on template/module/static/src/js/module_name.js:
```diff
@@ -1,7 +1,20 @@
+
+    core.bus.on('web_client_ready', null, function () {
+        // Script that will be loaded when document is ready
+    });
```
This approach doesn't work for `website`. Instead, the proper way to determine that the document is ready seems to be via `ready()` in `web_editor.base`, which returns a Promise object. This means that for `website`, the example here should look more like:

```
var base = require('web_editor.base');
base.ready().done(function() {
    // Script that will be loaded when document is ready
});
```

pedrobaeza replied:
This is not used in website, but on web client (as stated). Maybe we should include both in 2 files: one for web, and another for website.

lasley replied:
I see no reason to not include as many examples as possible. @pedrobaeza you're thinking a separate file would be good here instead of just putting both here and commenting the difference?

pedrobaeza replied:
Yes, I'm thinking about 2 files, naming then `web_module_name.js` and `website_module_name.js`

lasley commented:
Split website sample into separate file and added a web_ namespace to the web ones. 

I did not add a website tour file, because I am not quite sure of the intricacies between the back and front end tours (if any). IMO it would be bad for me to add a sample I am not 100% positive of 😄

gurneyalex commented:
:+1:

lasley commented:
Squashed. This is good for merge?
